kernel/syscall.c

```diff
@@ -3169,6 +3169,33 @@ long myst_syscall_mbind(
     return ret;
 }
 
+long myst_syscall_get_mempolicy(
+    int* mode,
+    unsigned long* nodemask,
+    unsigned long maxnode,
+    void* addr,
+    unsigned long flags)
+{
+    unsigned long size = 0;
+    myst_round_up(maxnode, sizeof(unsigned long) * 8, &size);
+    size /= 8;
+
+    long params[6] = {
+        (long)mode, (long)nodemask, (long)size, (long)addr, (long)flags};
+    long ret = myst_tcall(SYS_get_mempolicy, params);
+
+    // As Mystikos does not support mbind or set_mempolicy, we override nodemask
+    // returned by the target to indicate the thread is not allowed to specify a
+    // policy on any node in mbind or set_mempolicy when
+    // flags = MPOL_F_MEMS_ALLOWED
+    if (flags == 1 << 2)
+    {
+        *nodemask = 0;
```

Comment:
if `maxnode` > 64, more than one unsigned long element pointed to by `nodemask` need to be zeroed.

Reply:
updated implementation